install.sh

```diff
@@ -35,27 +58,51 @@ requirement() {
 }
 
 gen_config_file() {
-    SUFFIX=`tr -dc A-Za-z0-9 </dev/urandom | head -c 8`
-    API_KEY=`cscli bouncers add crowdsec-openresty-bouncer-${SUFFIX} -o raw`
-    API_KEY=${API_KEY} CROWDSEC_LAPI_URL="http://127.0.0.1:8080" envsubst < ./config/config_example.conf > "${CONFIG_PATH}crowdsec-openresty-bouncer.conf"
-    echo "New API key generated in config '${CONFIG_PATH}crowdsec-openresty-bouncer.conf'"
+    #Don't overwrite the existing file
+    if [ ! -f "${CONFIG_PATH}/crowdsec-openresty-bouncer.conf" ]; then
+        if [ -z ${DOCKER} ]; then
+            SUFFIX=$(tr -dc A-Za-z0-9 </dev/urandom | head -c 8)
+            API_KEY=$(cscli bouncers add "crowdsec-openresty-bouncer-${SUFFIX}" -o raw) 
+            API_KEY=${API_KEY} CROWDSEC_LAPI_URL="http://127.0.0.1:8080" envsubst < ./config/config_example.conf > "${CONFIG_PATH}/crowdsec-openresty-bouncer.conf"
+            echo "New API key generated to be used in '${CONFIG_PATH}/crowdsec-openresty-bouncer.conf'"
+        else
+            #Docker doesn't support envsubst by default
+            API_KEY="1234567890abcdef"
+            cp config/config_example.conf "${CONFIG_PATH}/crowdsec-openresty-bouncer.conf"
+            sed -i 's|API_KEY=.*|API_KEY='${API_KEY}'|' "${CONFIG_PATH}/crowdsec-openresty-bouncer.conf"
+            sed -i 's|CROWDSEC_LAPI_URL=.*|CROWDSEC_LAPI_URL="http://127.0.0.1:8080"|' "${CONFIG_PATH}/crowdsec-openresty-bouncer.conf"
+        fi
+    else
+        #Patch the existing file with new parameters if the need to be added
+        echo "Patch crowdsec-openresty-bouncer.conf .." 
+        sed "s/=.*//g" "${CONFIG_PATH}/crowdsec-openresty-bouncer.conf" > /tmp/crowdsec.conf.raw
+        sed "s/=.*//g" ./config/config_example.conf  > /tmp/config_example.conf.raw
+        if grep -vf /tmp/crowdsec.conf.raw /tmp/config_example.conf.raw ; then
+            grep -vf /tmp/crowdsec.conf.raw /tmp/config_example.conf.raw > /tmp/config_example.newvals
+            cp "${CONFIG_PATH}/crowdsec-openresty-bouncer.conf" "${CONFIG_PATH}/crowdsec-openresty-bouncer.conf.bak"
+            #Make sure we start on a new line.
+             echo "" >>"${CONFIG_PATH}/crowdsec-openresty-bouncer.conf"
+            grep -f /tmp/config_example.newvals /tmp/crowdsec/config/config_example.conf >> "${CONFIG_PATH}/crowdsec-openresty-bouncer.conf"
+        fi
+    fi
+    # Not sure why couldn't patch the path using envsubst
```

Comment:
Maybe there is no envsubst in the NPM docker image ? 
Don't forget to remove not important comments please.

Reply:
yea this make sense i'll update the docker image where this is required.